app/src/main/java/ch/epfl/balelecbud/WelcomeActivity.java

```diff
@@ -56,47 +57,32 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_welcome);
 
-        Button infoButton = findViewById(R.id.infoButton);
-        infoButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                openInfoActivity();
-            }
-        });
+        openActivity(FestivalInformationActivity.class, (Button) findViewById(R.id.infoButton));
+        openActivity(ScheduleActivity.class, (Button) findViewById(R.id.scheduleButton));
+        openActivity(MapViewActivity.class, (Button) findViewById(R.id.mapButton));
+        openActivity(TransportActivity.class, (Button) findViewById(R.id.transportButton));
+        openActivity(PointOfInterestActivity.class, (Button) findViewById(R.id.poiButton));
 
-        Button scheduleButton = findViewById(R.id.scheduleButton);
-        scheduleButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                openScheduleActivity();
-            }
-        });
-
-        Button mapButton = findViewById(R.id.mapButton);
-        mapButton.setOnClickListener(new View.OnClickListener() {
+        final Button signOutButton = findViewById(R.id.buttonSignOut);
+        signOutButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                openMapActivity();
+                signOut();
             }
         });
 
-        Button transportButton = findViewById(R.id.transportButton);
-        transportButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                openTransportActivity();
-            }
-        });
+        setUpLocation();
+    }
 
-        final Button signOutButton = findViewById(R.id.buttonSignOut);
-        signOutButton.setOnClickListener(new View.OnClickListener() {
+    private void openActivity(final Class activityToOpen, Button button) {
```

Comment:
The name of the `openActivity` method is misleading. It doesn't open an activity. It binds the button to an activity. An adequate name would be `bindButtonToActivity`